src/chainstate/burn/operations/peg_in.rs

```diff
@@ -423,7 +430,7 @@ mod tests {
         let addr_bytes = test::random_bytes(&mut rng);
         let stx_address = StacksAddress::new(1, addr_bytes.into());
         data.extend_from_slice(&addr_bytes);
-        data.extend_from_slice(&[0; 40]); // Padding contract name
+        data.push(0); // Contract name not provided, so size is 0
```

Comment:
You also need to test the following:

* a non-zero size length with no additional bytes for the name (this is invalid)
* a non-zero size length with insufficiently many bytes for the name (this is invalid)
* a non-zero size length with more than enough bytes for the name (this should be fine, but the name should be truncated to the given size, and the remaining bytes should be in the memo)
* a zero-size name with trailing bytes (this should be fine -- these bytes should all be in the memo)